### ISSUE TITLE
[FEATURE REQUEST] Add Deno Support

### ISSUE THREAD

**Issue by LePichu** [vuejs/language-tools#2258]
As of the time of writing, Vue compilation is supported on the Deno runtime by the main compiler using Vite but also other community powered tools which use the compiler in alternate ways to get Vue working with Deno.

The last piece remaining is proper IDE support, so if possible, please consider adding a runtime option in the configuration which can be set to Deno and make Volar co-operate with Deno Language Server instead.

Currently, it almost works but does not recognize Deno Import Maps and often TypeScript files if they use Deno-specific things.

TL;DR - Please add functioning auto-complete for Deno runtime users which picks imports from Deno's import maps or URLs directly.

**Comment by rusty-easy** [comment]
As of today, Deno imports still don't work.

**Comment by theoriginalstove** [comment]
I think this feature would be great, especially now that Deno v2 has backwards compatibility with node.

**Comment by HYliangkai** [comment]
At present, projects using deno+vite+vue can run normally, but what is missing now is LSP support

**Comment by not-Ryan** [comment]
It could also be a fix for something like #5063 in huge projects? 🤷

**Comment by johnsoncodehk** [comment]
We are now fully integrated with tsserver and implement TS support based on the [TypeScript Language Service Plugin](https://github.com/microsoft/TypeScript/wiki/Writing-a-Language-Service-Plugin), so this request is no longer valid.

**Comment by bsdayo** [comment]
At this time, Vue SFCs still have no type suggestions with Deno runtime.

Repro: https://github.com/denoland/tutorial-with-vue

With `"deno.enabled": false` in `.vscode/settings.json`, working:

<img width="768" height="396" alt="Image" src="https://github.com/user-attachments/assets/d94d6ed1-be8c-42de-ba07-a782c4e49866" />

With `"deno.enabled": true`, not working (no suggestions)

<img width="768" height="396" alt="Image" src="https://github.com/user-attachments/assets/7d7f48be-5a43-4251-bd5a-924a9f72242f" />

Building is OK

**Comment by HYliangkai** [comment]
A year has passed, and there is no solution to this problem yet...

**Comment by LePichu** [comment]
> A year has passed, and there is no solution to this problem yet...

That won't ever change, it is not a priority for neither Volar nor Deno to actually provide Deno support, which is ironic because Deno wanted to be the one-stop shop for everything JavaScript, as for Volar Team, they have better issues to solve and make the experience better for Node itself, which is fine.
